### PR TITLE
[Check-in] Add section for the Logout Endpoint

### PR DESCRIPTION
**Pull request by nikosev**
# Summary

This PR introduces the following changes:

- Add section for Logout Endpoint
- Fix case of OIDC terminology
- Update instruction of creating SAML SP metadata
- Add section for Redirect URIs
- Add section for UserInfo Endpoint
- Add section for Introspection Endpoint
- Fix notes refering to the endpoints table
- Move flow descriptions from `Endpoints` to `Grant Types` section
- Make duplicate headers unique in order to use them in links

### CODE REVIEW

**Review by NicolasLiampotis — APPROVED** [review]
LGTM

**Review by vardizzo-lab — APPROVED** [review]

File: content/en/providers/check-in/sp/_index.md [hunk 1]
```diff
@@ -819,17 +853,281 @@ curl -X POST "${TOKEN_ENDPOINT}" \
...
+
+It is recommended that the request use the HTTP `GET` method and the Access
+Token be sent using the Authorization header field.
+
+Example request:
+
+```shell
+curl -X GET ${USERINFO_ENDPOINT}?
+  -H "Content-type: application/json"
+  -H "Authorization: Bearer ${ACCESS_TOKEN}" | python -m json.tool;
```
Comment (gwarf) [inline content/en/providers/check-in/sp/_index.md:984]:
```suggestion
$ curl -X GET ${USERINFO_ENDPOINT}?
    -H "Content-type: application/json"
    -H "Authorization: Bearer ${ACCESS_TOKEN}" | python -m json.tool;
```

File: content/en/providers/check-in/sp/_index.md [hunk 2]
```diff
@@ -819,17 +853,281 @@ curl -X POST "${TOKEN_ENDPOINT}" \
...
+The protected resource calls the introspection endpoint using an HTTP `POST`
+request with parameters sent as `application/x-www-form-urlencoded`.
+
+Example request:
+
+```shell
+curl -X POST "${INTROSPECTION_ENDPOINT}" \
+  -u "${CLIENT_ID}":"${CLIENT_SECRET}" \
+  -H "Content-Type: application/x-www-form-urlencoded" \
+  -d "token=${ACCESS_TOKEN}" | python -m json.tool;
```
Comment (gwarf) [inline content/en/providers/check-in/sp/_index.md:1033]:
```suggestion
$ curl -X POST "${INTROSPECTION_ENDPOINT}" \
    -u "${CLIENT_ID}":"${CLIENT_SECRET}" \
    -H "Content-Type: application/x-www-form-urlencoded" \
    -d "token=${ACCESS_TOKEN}" | python -m json.tool;
```

File: content/en/providers/check-in/sp/_index.md [hunk 3]
```diff
@@ -819,17 +853,281 @@ curl -X POST "${TOKEN_ENDPOINT}" \
...
+  in the configuration of the Service in
+  [EGI Federation Registry](https://aai.egi.eu/federation). Note that you need
+  to include either the `client_id` or `id_token_hint` parameter in case the
+  `post_logout_redirect_uri` is included.
+
+##### Example Request
+
+```shell
+curl -X GET ${LOGOUT_ENDPOINT}?
+  id_token_hint=${ID_TOKEN}
```
Comment (gwarf) [inline content/en/providers/check-in/sp/_index.md:1106]:
```suggestion
$ curl -X GET "${LOGOUT_ENDPOINT}?id_token_hint=${ID_TOKEN}"
```

File: content/en/providers/check-in/sp/_index.md [hunk 4]
```diff
@@ -819,17 +853,281 @@ curl -X POST "${TOKEN_ENDPOINT}" \
...
+
+{{% alert title="Note" color="info" %}} You can find the `LOGOUT_ENDPOINT` in
+the [Endpoints](#endpoints) table.{{% /alert %}}
+
+##### Example Request with redirection
+
+```shell
+curl -X GET ${LOGOUT_ENDPOINT}?
+  post_logout_redirect_uri=${POST_LOGOUT_REDIRECT_URI}&
+  client_id=${CLIENT_ID}
```
Comment (gwarf) [inline content/en/providers/check-in/sp/_index.md:1117]:
```suggestion
$ curl -X GET "${LOGOUT_ENDPOINT}post_logout_redirect_uri=${POST_LOGOUT_REDIRECT_URI}&client_id=${CLIENT_ID}"
```

**Review by gwarf — APPROVED** [review]
LGTM, did some small fixes on formatting of excertps.

**Comment by gwarf** [comment]
@nikosev please confirm the changes I made are fine and I will merge, thanks!

**Comment by nikosev** [comment]
Hi @gwarf !
I thought that the dollar sign (`$`) in front of the commands should be placed when the command is followed by the output of the command. Since the curl commands are now followed by their output, I didn't include the dollar sign.
If we need to add the dollar sign to all commands, then I need to update all the commands in the SP guide.

Also, the indentation of all the curl commands is set to 2 spaces. Should I update it to 4 spaces?

**Comment by gwarf** [comment]
Hi, so yes, we agreed to use the prompt sign on all commands, to have a coherent usage.
We have some info on https://docs.egi.eu/about/contributing/style/#basic-rules and https://docs.egi.eu/about/contributing/shortcodes/#code-or-shell-snippets.

So for the indentation of the commands, basically they should always be two spaces from the parent/reference. Here there was the need to increment them due to adding the `$ ` prompt, so that the second line is properly padded.

**Review by gwarf — CHANGES_REQUESTED** [review]
Looks good, thanks!
But please review/update shell excerpts.

**Comment by nikosev** [comment]
@gwarf I have made some changes based on our discussion. If there is anything that needs improvement, please let me know.

File: content/en/providers/check-in/sp/_index.md [hunk 5]
```diff
@@ -483,42 +461,96 @@ The request parameters of the Authorization endpoint are:
...
   used. For **Authorization Code** grant set `response_type=code`. This way the
-  response will include an authorization code.
+  response will include an Authorization Code.
 
-#### Token Endpoint
+Example request:
 
-To obtain an Access Token, an ID Token, and optionally a Refresh Token, the
-Client sends a Token Request to the Token Endpoint.
+```shell
```
[the remark this answers is not in the document]
Comment (nikosev) [reply content/en/providers/check-in/sp/_index.md]: I converted the syntax highlighter to `http`. I wanted to copy the code block as it is in the [OpenID Connect core specification](https://openid.net/specs/openid-connect-core-1_0.html#AuthResponse)

File: content/en/providers/check-in/sp/_index.md [hunk 6]
```diff
@@ -346,24 +364,20 @@ To find the ID and secret of your client, do the following:
...
-   of the overview table or **Edit** the particular client and look for the
-   **Redirect URI(s)** in the **Protocol** tab.
-
-#### Setting additional information about the client
-
-It is strongly suggested that you add a short **description** and a **logo** for
-the client. Lastly, you need to set the email addresses of one or more contacts.
+### Redirection URIs
+
+OpenID Connect Services MUST pre-register one or more Redirection URI(s) to
```
Comment (gwarf) [inline content/en/providers/check-in/sp/_index.md:369]:
```suggestion
OpenID Connect Services **MUST** pre-register one or more Redirection URI(s) to
```

File: content/en/providers/check-in/sp/_index.md [hunk 7]
```diff
@@ -346,24 +364,20 @@ To find the ID and secret of your client, do the following:
...
+### Redirection URIs
+
+OpenID Connect Services **MUST** pre-register one or more Redirection URI(s) to
+which authentication responses from EGI Check-in will be sent. EGI Check-in
+utilises exact matching of the redirect URI specified in an authentication
+request against the pre-registered URIs
+[OAuth2-BCP](https://datatracker.ietf.org/doc/html/draft-ietf-oauth-security-topics#section-4.1),
+with the matching performed as described in
+[RFC3986](https://www.rfc-editor.org/rfc/rfc3986#section-6.2.1) (Simple String
+Comparison). Redirection URIs MUST use the schemata defined in Section 3.1.2.1
```
Comment (gwarf) [inline content/en/providers/check-in/sp/_index.md:376]:
```suggestion
Comparison). Redirection URIs **MUST** use the schemata defined in Section 3.1.2.1
```

File: content/en/providers/check-in/sp/_index.md [hunk 8]
```diff
@@ -346,24 +364,20 @@ To find the ID and secret of your client, do the following:
...
+of the
+[OIDC-Core](https://openid.net/specs/openid-connect-core-1_0.html#AuthRequest)
+specification. Note that the Redirection URI MUST use the `https` scheme; the
```
Comment (gwarf) [inline content/en/providers/check-in/sp/_index.md:379]:
```suggestion
specification. Note that the Redirection URI **MUST** use the `https` scheme; the
```

File: content/en/providers/check-in/sp/_index.md [hunk 9]
```diff
@@ -244,19 +244,37 @@ the name in other languages which are commonly used in the geographic scope of
...
+  - an `<md:ContactPerson>` element with a contactType of `support` and/or a
+    `<md:ContactPerson>` element with a contactType of `technical`. The
+    `<md:ContactPerson>` element(s) should contain at least one
+    `<md:EmailAddress>`. The support address may be used for generic support
+    questions about the service, while the technical contact may be contacted
+    regarding technical interoperability problems. The technical contact must be
+    responsible for the technical operation of the service represented by your
+    SP.
+
+If the SP supports the Single Logout profile, then its metadata MUST contain
```
Comment (gwarf) [inline content/en/providers/check-in/sp/_index.md:269]:
```suggestion
If the SP supports the Single Logout profile, then its metadata **MUST** contain
```

File: content/en/providers/check-in/sp/_index.md [hunk 10]
```diff
@@ -483,42 +461,96 @@ The request parameters of the Authorization endpoint are:
...
+    &state=af0ifjsldkj
+```
+
+##### Token Request
+
+A Client makes a Token Request by presenting its Authorization Grant (in the
+form of an Authorization Code) to the Token Endpoint using the `grant_type`
+value `authorization_code`, as described in Section 4.1.3 of OAuth 2.0
+[RFC6749](https://www.rfc-editor.org/rfc/rfc6749#section-4.1.3). If the Client
+is a Confidential Client, then it MUST authenticate to the Token Endpoint using
```
Comment (gwarf) [inline content/en/providers/check-in/sp/_index.md:496]:
```suggestion
is a Confidential Client, then it **MUST** authenticate to the Token Endpoint using
```

**Review by gwarf — APPROVED** [review]

**Comment by gwarf** [comment]
Perfect, thanks a lot @nikosev, let's merge!

**Pull request merged**